cli/pcluster/utils.py

```diff
@@ -105,22 +107,31 @@ def paginate_boto3(method, **kwargs):
             yield result
 
 
+def generate_random_bucket_name(bucket_name_prefix):
+    """
+    Generate a random bucket name, with the given prefix.
+
+    Bucket name must be at least 3 and no more than 63 characters long.
+    """
+    random_string = "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(16))
```

Comment:
Could you please add a small comment / example of generated bucket name? Is this mechanism safe enough to avoid name collisions?

Reply:
Example added.
This random generation is using the cluster name as prefix and adding a random string of 16 characters.

Clearly a name collision could happen, in that case the entire stack creation will fail logging an error and the exception, before this patch it was failing without a clear message for the user.

Reply:
it's not perfect but if you only consider the randomly generated substring (last 16 chars) we have the permutation of 36 unique chars (string.ascii_lowercase + string.digits) over 16 positions which gives us 7.9586611099e+24 unique strings and if we assume that the random function has a uniform distribution (which is not completely true but I think we are quite close to that) then probability of a collision is really low.
If we think this is not enough we could handle the error code saying that the bucket name is already in use and generate a different name.